### DISCUSSION TITLE
Only call the function and not run the whole script

### DISCUSSION THREAD

**Discussion by techierishi** [yuin/gopher-lua#446] [Q&A] [answered]
I want the know if there is a way to only call the function present in the Lua script. I don't want rest of the script to run.

for example:

test.lua
```lua
function test()
    return "test string"
end
  
print("Hello World!")
```

Now in following go code whole above lua script is running including the print statement. I just want `test` function to be executed

```lua
L := lua.NewState()
L.DoFile(filePath)

defer L.Close()

luaFn := L.GetGlobal("test")
if luaFn == lua.LNil {
return nil, fmt.Errorf("function not found")
}

err := L.CallByParam(lua.P{
Fn:      luaFn,
NRet:    1,
Protect: true,
})
```

**Comment by cloudwindy** [comment, marked as answer]
I don't think it's possible, the whole script have to be executed to get the function. Imagine this situation:
```
local b
function a()
  print(b)
end
b = 123
```
Any part of the code can effectively change how the behavior of that function.
